### DISCUSSION TITLE
hmr and nginx reverse proxy

### DISCUSSION THREAD

**Discussion by ad-on-is** [nuxt/nuxt#16602] [Questions]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.17.0`
- Nuxt Version:     `3.0.0-rc.4`
- Nitro Version:    `0.4.24`
- Package Manager:  `pnpm@7.13.5`
- Builder:          `vite`
- User Config:      `css`, `ssr`, `modules`, `build`
- Runtime Modules:  `@nuxtjs/tailwindcss@5.3.5`, `@pinia/nuxt@0.3.1`
- Build Modules:    `-`
------------------------------

## Nginx

I've stetup nginx proxy in a way, so it proxies all requests based on the request uri.

- `https://local-1234.example.com` proxies the request to a any server running on port `1234`
- etc..

Running the nuxt app on port `3000` and accessing it via `https://local-3000.example.com` fails to load MHR properly, and throws the following error:

GET https://local-3000.add.dnmc.in:24678/_nuxt/ net::ERR_SSL_PROTOCOL_ERROR

### Describe the bug

I tried the following mentioned solution: https://github.com/vitejs/vite/issues/7849#issuecomment-1106346538

```
vite: {
    server: {
      hmr: {
        clientPort: 443,
      },
    },
  },
```

but this still does not work and I get the following errors

WebSocket connection to 'wss://local-3000.add.dnmc.in/_nuxt/' failed: setupWebSocket @ client:239(anonymous) @ client:233
​ [vite] server connection lost. polling for restart...​          
GET https://local-3000.add.dnmc.in/_nuxt/ net::ERR_ABORTED 404
waitForSuccessfulPing @ client:435 (anonymous) @ client:257

```

### Additional context

_No response_

### Logs

_No response_

**Comment by HendrikJan** [comment]
Did you find a solution to run Nuxt hmr behind reverse proxy?

**Reply by maou-shonen** [reply]
> Did you find a solution to run Nuxt hmr behind reverse proxy?

```ts
   vite: {
    server: {
      hmr: {
        clientPort: 80, this port is used on the browser, usually you will want to be consistent with nuxt
        port: 3001, // this port must be different from nuxt
      },
    },
  },
```
In the example above,
Then use a reverse proxy server to reverse proxy `:80/_nuxt/` to `:3001`, and rest of the traffic to `:3000`
Remember hmr must use WS

**Reply by HendrikJan** [reply]
Thanks 👍. I'll try that.
